api/rest/restapi.go

```diff
@@ -317,6 +434,9 @@ func (api *API) Shutdown() error {
 // requests.
 func (api *API) SetClient(c *rpc.Client) {
 	api.rpcClient = c
+
+	// One notification for http server and one for libp2p server.
+	api.rpcReady <- struct{}{}
```

Comment:
I think this will block indefinitely in the case where the api starts up with only an http server or only a libp2p server

Reply:
no, I increased channel capacity to two elements.